### ISSUE TITLE
Proposal: add prefer-expect-type-of rule

### ISSUE THREAD

**Issue by y-hsgw** [vitest-dev/eslint-plugin-vitest#741]
## Summary

I propose adding a rule that recommends using Vitest’s type-testing matcher [expectTypeOf](https://vitest.dev/api/expect-typeof) instead of runtime typeof checks inside expect(...).

## Examples
Incorrect
```ts
expect(typeof 12).toBe("number")
expect(typeof "name").toBe("string")
expect(typeof true).toBe("boolean")
... etc
```

Correct
```ts
expectTypeOf(12).toBeNumber()
expectTypeOf("name").toBeString()
expectTypeOf(true).toBeBoolean()
...etc
```

**Comment by veritem** [comment]
a pr is always welcome :D

**Comment by sheremet-va** [comment]
Is there any reasoning for that? Runtime checks and type checks serve different purposes `expect(typeof)` is not the same as `expectTypeOf`. If anything, this should recommend `expect(value).toBeTypeOf(type)`

**Comment by y-hsgw** [comment]
Yeah, you're right — using `toBeTypeOf` would be more appropriate here.

I missed that perspective when I opened this issue. Sorry about that, and thanks for pointing it out!
